core/config.go

```diff
@@ -21,455 +21,146 @@ package core
-func (ngc NutsGlobalConfig) ServerAddress() string {
-	return ngc.v.GetString(addressFlag)
-}
-
-// InStrictMode helps to safeguard settings which are handy and default in development but not safe for production.
-func (ngc NutsGlobalConfig) InStrictMode() bool {
-	return ngc.v.GetBool(strictModeFlag)
-}
+	ngc.configMap = koanf.New(defaultDelimiter)
+	f := ngc.getConfigfile(cmd)
 
-// Load sets some initial config in order to be able for commands to load the right parameters and to add the configFile Flag.
-// This is mainly spf13/viper related stuff
-func (ngc *NutsGlobalConfig) Load(cmd *cobra.Command) error {
-	ngc.v.SetEnvPrefix(ngc.Prefix)
-	ngc.v.AutomaticEnv()
-	ngc.v.SetEnvKeyReplacer(strings.NewReplacer(ngc.Delimiter, "_"))
-	flagSet := pflag.NewFlagSet("config", pflag.ContinueOnError)
-	flagSet.String(configFileFlag, ngc.DefaultConfigFile, "Nuts config file")
-	flagSet.String(loggerLevelFlag, defaultLogLevel, "Log level (trace, debug, info, warn, error)")
-	flagSet.String(addressFlag, defaultAddress, "Address and port the server will be listening to")
-	flagSet.Bool(strictModeFlag, false, "When set, insecure settings are forbidden.")
-	flagSet.String(identityFlag, "", "Vendor identity for the node, mandatory when running in server mode. Must be in the format: urn:oid:"+NutsVendorOID+":<number>")
-	cmd.PersistentFlags().AddFlagSet(flagSet)
+	// load file
+	p := file.Provider(f)
+	if err = ngc.configMap.Load(p, yaml.Parser()); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return
+		}
+	}
 
-	// Bind config flag
-	// Bind log level flag
-	ngc.bindFlag(flagSet, configFileFlag)
-	ngc.bindFlag(flagSet, loggerLevelFlag)
-	ngc.bindFlag(flagSet, addressFlag)
-	ngc.bindFlag(flagSet, strictModeFlag)
-	ngc.bindFlag(flagSet, identityFlag)
+	// load env
+	e := env.Provider(defaultPrefix, defaultDelimiter, func(s string) string {
+		return strings.Replace(strings.ToLower(
+			strings.TrimPrefix(s, defaultPrefix)), "_", defaultDelimiter, -1)
+	})
+	// errors can't occur for this provider
+	_ = ngc.configMap.Load(e, nil)
 
-	// load flags into viper
-	pfs := cmd.PersistentFlags()
-	pfs.ParseErrorsWhitelist.UnknownFlags = true
-	if err := pfs.Parse(os.Args[1:]); err != nil {
-		if err != pflag.ErrHelp {
-			return err
-		}
+	// load cmd params
+	if len(os.Args) > 1 {
+		_ = cmd.PersistentFlags().Parse(os.Args[1:])
 	}
+	// errors can't occur for this provider
+	_ = ngc.configMap.Load(posflag.Provider(cmd.PersistentFlags(), defaultDelimiter, ngc.configMap), nil)
 
-	// load configFile into viper
-	if err := ngc.loadConfigFile(); err != nil {
-		return err
+	// load into struct
+	if err = ngc.configMap.Unmarshal("", ngc); err != nil {
+		return
 	}
 
 	// initialize logger, verbosity flag needs to be available
-	level, err := log.ParseLevel(ngc.v.GetString(loggerLevelFlag))
-	if err != nil {
-		return err
+	if _, err = log.ParseLevel(ngc.Verbosity); err != nil {
```

Comment:
You still have to set the level?

Reply:
I think that should be part of #40

Reply:
Then why do we have the option if it doesn't do anything?

Reply:
I'll set it here with a todo